library/kotlin/io/envoyproxy/envoymobile/stats/Tag.kt

```diff
@@ -0,0 +1,5 @@
+package io.envoyproxy.envoymobile
+
+/** Tag represents the [key,value] pair component of a time series name. */
+
+class Tag(internal val key: String, internal val value: String)
```

Comment:
The choice was between having a dedicated `Tag` class and pass around tags as `List<Tag>` or directly use a `Map<String, String>`. 

Decided to have a `Tag` class: it makes code easier to read for the public APIs. It will also make it easier to add enforcements to tags if needed down the line.

Reply:
I had written a comment about this before I got to read this. I appreciate the use of a clear class here, but I would like to see if we can do it in a way that pushes its use all the way to the bridge layer. i.e, we are converting from one thing to another before we _need_ to, and the conversion is not free. 

I'd like to hear @rebello95 and @buildbreaker opinion on the balance of the exposed API vs adding more conversion overhead.

Reply:
What if we just have a `Tags` class and have this class just be the `List<Tag>` internally and it can expose a nicer API? This way, you can also unify the `Stats.convert` method into this Tags class.

Reply:
Having a `Tags` class sounds a good option to me. Would like to hear more input too.

Reply:
@junr03 about the layers for the conversion:

there are two conversions: 1) from `Tag` or `Tags` to `Map<String, String>`, and 2) from `Map<String, String>` to native. The second conversion: From Map<String, String>` to native is done currently in `EnvoyEngineImpl` - same as headers. 

For the first conversion, I was thinking about pushing it down to the `EnvoyEngineImpl` and have it done along with the second conversion, it will require `EnvoyEngineImpl` to import `Tag` or `Tags` class, and I thought at the time it probably was not a good idea.

Reply:
Commented above; I don't see the need for having a Map intermediary.

Reply:
Similar to Headers, you can have a Tags class that allows for additional functionality and cleaner API. But it stores tags internally in a map<string, string>. That way there's no conversation or copy when passing it down.

Reply:
If it starts as a map and stays as a map, that's fine with me.

Reply:
sounds good to me